src/pykeen/utils.py

```diff
@@ -1777,6 +1780,63 @@ def isin_many_dim(elements: torch.Tensor, test_elements: torch.Tensor, dim: int
     return counts[inverse[: elements.shape[dim]]] > 1
 
 
+def minmax_normalize(array: np.ndarray) -> np.ndarray:
+    """
+    Normalize the given array using min-max normalization.
+
+    :param array:
+        array to normalize
+
+    :return:
+        normalized array
+    """
+    max_lit, min_lit = np.max(array, axis=0), np.min(array, axis=0)
+    return (array - min_lit) / (max_lit - min_lit + 1e-8)
+
+
+def filter_triples_by_relations(
+    triples: LabeledTriples, min_occurrences: int = 5, regex: str = r"^(?!.*http://rdf.freebase.com/key/).*$"
```

Comment:
can we rewrite this to work on the `mapped_triples` instead? Working on large arrays of strings is slow compared to integer arrays.

Reply:
If we do that, it will break the sequence of the mapped data relation IDs because some relations may be removed. But with the current implementation, the triples are first filtered, and the relation-to-ID mapping happens later in `create_matrix_of_literals()` thus guaranteeing that the IDs are sequential.